### PR TITLE
[SIG-3233] Sharable ESLint configuration

### PR DESCRIPTION
**Pull request by jpoppe**
Todo before merge: run fix on a few comma dangles, check GitHub integration

* Lint rules have been separated and moved to it's own plugin
* Typescript ready :)
* Cleaned and optimized configuration
* Reimplemented and fixed 'broken' lint-staged plugin
* Added Jest plugin
* Added Node plugin
* Added find rules script
* Linting now also checks additonal scripts
* Removed clashing prettier/airbnb plugins (packages will be removed later) 
* Made clear separation and proposed rules really clear, needs to be picked up as chores? Out of scope for now
* Plugin could be shared with Amstedam organization but needs to be discussed
* Much more :)

### CODE REVIEW

File: package.json
Comment (aditudorache) [inline package.json:9]: Is this version of node necessary for eslint? I think it might conflict with gihub actions workflow where the used version is 12.8

File: .eslintrc.yaml
Comment (aditudorache) [inline .eslintrc.yaml:1]: I noticed that this file is `yaml` while some others configuration files are `js`. Shouldn't we keep one file type for consistency?

File: package.json
Comment (aditudorache) [inline package.json:9]: Is this version of node necessary? It doesn't match the one used in the github actions (12.8) can we sync that?

**Review by aditudorache — COMMENTED** [review]
LGTM; I have a few questions and I saw there are few `comma-dangle` errors when running the linter.

**Comment by aditudorache** [comment]
Can we target this to the `post-release` branch before merging?

**Review by boris-arkenaar — COMMENTED** [review]
So many rules 😱 
Can't we just stick to an industry-standard as much as possible? Like `airbnb`.

File: package.json
Comment (jpoppe) [reply package.json:9]: No this should not be there, nice catch! thanks!

**Comment by jpoppe** [comment]
> So many rules 
> Can't we just stick to an industry-standard as much as possible? Like `airbnb`.

No, you will miss about 70% of the rules (many of those rules prevent discussions here), `airbnb` was where the project started.

File: .eslintrc.yaml
Comment (jpoppe) [reply .eslintrc.yaml:1]: The YAML files is the application configuration and should be as good as empty in the ideal world, the js files are part of the `eslint-plugin-amsterdam` which could eventually live somewhere out of this repository.

Comment (aditudorache) [reply .eslintrc.yaml:1]: Not sure I understand the explanation, why don't we keep the same format for eslint configuration? It doesn't really matter which format as long is consistent overall. I would choose js because most of the projects at `datapunt` are using this format. As a matter of fact, this is the original format before this PR.

File: eslint-plugin-amsterdam/package.json
Comment (aditudorache) [inline eslint-plugin-amsterdam/package.json:5]: Can we use `Datapunt Amsterdam` here? maintainers names can be put in a MAINTAINERS.md file if needed

**Comment by boris-arkenaar** [comment]
> > So many rules
> > Can't we just stick to an industry-standard as much as possible? Like `airbnb`.
> 
> No, you will miss about 70% of the rules (many of those rules prevent discussions here), the project started with `airbnb` rules. which were not sufficient and left to much room for discussion on GitHub reviews.

Got it!

File: eslint-plugin-amsterdam/package.json
Comment (jpoppe) [reply eslint-plugin-amsterdam/package.json]: I removed the author field for now, we could add add Datapunt or something else if this plugin will ever be published.

**Review by MauritsR — APPROVED** [review]
Looks good :)

File: .eslintrc.yaml
Comment (boris-arkenaar) [reply .eslintrc.yaml:1]: I'm also still interested to hear your reasoning for choosing the yaml format. :)

Comment (aditudorache) [reply .eslintrc.yaml:1]: @boris-arkenaar, mee too. @jpoppe, we agreed to talk about this PR next week. I think this might be merged by mistake. Could you please revert this?

**Pull request merged**
